src/internal/SlidingSheet.js

```diff
@@ -0,0 +1,439 @@
+      const dSecondaryAbs = Math.abs(currentSecondary - this.touchStartSecondary);
+      // If the user has moved his thumb ten pixels in either direction,
+      // we can safely make an assumption about whether he was intending
+      // to swipe or scroll in the horizontal case
+      const threshold = 10;
+
+      if (dPrimaryAbs > threshold && dSecondaryAbs <= threshold) {
+        event.preventDefault();
+        this.swipeStartPrimary = currentPrimary;
+        this.setState({
+          swiping: this.props.open ? 'closing' : 'opening',
+        });
+        this.setPosition(this.getTranslatePrimary(currentPrimary));
+      } else if (dPrimaryAbs <= threshold && dSecondaryAbs > threshold) {
+        this.onBodyTouchEnd();
+      }
+    }
+  };
+
+  onBodyTouchEnd = (event) => {
+    if (this.state.swiping) {
+      const vertical = this.isVertical();
+      const currentPrimary = vertical ? event.changedTouches[0].pageY : event.changedTouches[0].pageX;
+      const translateRatio = this.getTranslatePrimary(currentPrimary) / this.getMaxTranslatePrimary();
+
+      this.maybeSwiping = false;
+      const swiping = this.state.swiping;
+      this.setState({
+        swiping: null,
+      });
+
+      // We have to open or close after setting swiping to null,
+      // because only then CSS transition is enabled.
+      if (translateRatio > 0.5) {
+        if (swiping === 'opening') {
+          this.setPosition(this.getMaxTranslatePrimary());
+        } else {
+          this.close('swipe');
+        }
+      } else {
+        if (swiping === 'opening') {
+          this.open('swipe');
+        } else {
+          this.setPosition(0);
+        }
+      }
+    } else {
+      this.maybeSwiping = false;
+    }
+
+    document.body.removeEventListener('touchmove', this.onBodyTouchMove);
+    document.body.removeEventListener('touchend', this.onBodyTouchEnd);
+    document.body.removeEventListener('touchcancel', this.onBodyTouchEnd);
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      containerClassName,
+      containerStyle,
+      direction,
+      modal,
+      overlayClassName,
+      overlayStyle,
+      style,
+      zDepth,
+    } = this.props;
+
+    const styles = this.getStyles();
+
+    let overlay;
+    if (modal) {
+      overlay = (
+        <Overlay
+          ref="overlay"
+          show={this.shouldShow()}
+          className={overlayClassName}
+          style={Object.assign(styles.overlay, overlayStyle)}
+          transitionEnabled={!this.state.swiping}
+          onTouchTap={this.handleTouchTapOverlay}
+        />
+      );
+    }
+
+    return (
+      <div
+        className={className}
+        style={Object.assign({width: '100%'}, style)}
+      >
+        <EventListener target="window" onKeyUp={this.handleKeyUp} />
+        {overlay}
+        <Paper
+          ref="clickAwayableElement"
+          zDepth={zDepth}
+          rounded={false}
+          transitionEnabled={!this.state.swiping}
```

Comment:
Isn't the point of disabling transitions when swiping that while we're swiping we want to manually set the height when the user moves their finger, and transitions would create a delay between the user's action and the UI responding? Or are you referring to something else?